include/CL/sycl/buffer/detail/buffer.hpp

```diff
@@ -25,6 +25,20 @@ namespace cl {
 namespace sycl {
 namespace detail {
 
+/** Used to retrieve the type return by an Iterator,
+ *  in particular a const-iterator<T> returns const T.
+ *  The attribute is_const is true iff the iterator is
+ *  a const one
+ */
+template<typename Iterator>
+struct iterator_value_type
+{
+  using value_type  = typename std::remove_pointer_t<
```

Comment:
Yes, 
`std::iterator_traits<Iterator>::value_type `
with Iterator being a `const-iterator<T>` returns `T` instead of `const T`